src/modules/flutterwave/flutterwave.service.spec.ts

```diff
@@ -0,0 +1,116 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { FlutterwaveService } from './flutterwave.service';
+import { HttpService } from '@nestjs/axios';
+import { ConfigService } from '@nestjs/config';
+import { Repository } from 'typeorm';
+import { Payment, PaymentStatus } from './entities/payment.entity';
+import { getRepositoryToken } from '@nestjs/typeorm';
+import { of } from 'rxjs';
+import { AxiosResponse } from 'axios';
+import { CreateFlutterwavePaymentDto } from './dto/create-flutterwavePaymentDto';
+
+describe('FlutterwaveService', () => {
+  let service: FlutterwaveService;
+  let httpService: HttpService;
+  let paymentRepo: Repository<Payment>;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        FlutterwaveService,
+        {
+          provide: HttpService,
+          useValue: {
+            get: jest.fn(),
+            post: jest.fn(),
+          },
+        },
+        {
+          provide: ConfigService,
+          useValue: {
+            get: jest.fn().mockReturnValue('mock-base-url'),
+          },
+        },
+        {
+          provide: getRepositoryToken(Payment),
+          useClass: Repository,
+        },
+      ],
+    }).compile();
+
+    service = module.get<FlutterwaveService>(FlutterwaveService);
+    httpService = module.get<HttpService>(HttpService);
+    paymentRepo = module.get<Repository<Payment>>(getRepositoryToken(Payment));
+  });
+
+  it('should initiate a payment successfully', async () => {
+    // Mocking the payment plan response
+    const paymentPlanResponse: AxiosResponse = {
```

Comment:
Remove comments. Besides this is you assigning a value to a variable not mocking